### DISCUSSION TITLE
Why the config file is not in ~/.mc/ ?

### DISCUSSION THREAD

**Discussion by 160610101** [minio/minio#11657] [Q&A] [answered]
Hi, there. I cannot find the `config.json` correctly.

I use `$ docker run -it --entrypoint=/bin/sh minio/mc` to run, and `# /usr/bin/mc config host add minio_docker $AWS_S3_ENDPOINT_URL $MINIO_ACCESS_KEY $MINIO_SECRET_KEY;` to add a host. Everything seems to be ok:

> mc: Configuration written to `/root/.mc/config.json`. Please update your access credentials.
mc: Successfully created `/root/.mc/share`.
mc: Initialized share uploads `/root/.mc/share/uploads.json` file.
mc: Initialized share downloads `/root/.mc/share/downloads.json` file.
Added `minio_docker` successfully.

But there is not a directory named `/root/.mc/`.  Well, it says in README.md that `mc` stores all its configuration information in `~/.mc/config.json` file. But I didn't find it here either.
Last, I find some directories named `.mc` as follows, and there are `config.json`, `config.json.old`, `share/`, etc in them.  It seems every time I run it, I get a directory like this.

> /var/snap/docker/common/var-lib-docker/overlay2/4a29ee17c0b2cb5199d14a2525648bd25ba330e3f20019ec3e68c3d71ed33b81/diff/root/.mc

> /var/snap/docker/common/var-lib-docker/overlay2/f608c655759525a19e2d00bf6061be4f14bef337cc8d8a7455f6d7ad1f1c369b/diff/root/.mc

> /var/snap/docker/common/var-lib-docker/overlay2/72bfaa81765598d6dfcc9b8703747af0689b9df78323b2b64b71bef712f53761/diff/root/.mc

Where is the latest `config.json`? Why are they here instead of in their normal position?
Thanks.

**Comment by harshavardhana** [comment, marked as answer]
Just ran 

```
docker run -it --entrypoint=/bin/sh minio/mc
```

```
docker inspect 15cc3e47e247 | jq .[].GraphDriver.Data
{
  "LowerDir": "/var/lib/docker/overlay2/b5fec18c9bfec606e6c667e1c40cfe5f6abecb8e9fc968df8002321c02f4cbbf-init/diff:/var/lib/docker/overlay2/c2579148cfb5ecf39a912de188f184b0319c650b6d2e37f3f9465b1a04808397/diff:/var/lib/docker/overlay2/653a5e58a161de4ecbfd7a6a99c4ca9431d403d096d607094622fba0fee3c076/diff:/var/lib/docker/overlay2/e98a8cc4953ca2de832aed6deb4018438219c85459c4d335c44ba6c56972a12b/diff:/var/lib/docker/overlay2/3fa6b0a07fc9a311147aa38bbaac40cef03b5ec99f3f0687697534a6cf3d75ea/diff:/var/lib/docker/overlay2/cd87db5a3edd0891c8315c69cf6494898967d1f38b02d99bc0b415c454a8a36b/diff",
  "MergedDir": "/var/lib/docker/overlay2/b5fec18c9bfec606e6c667e1c40cfe5f6abecb8e9fc968df8002321c02f4cbbf/merged",
  "UpperDir": "/var/lib/docker/overlay2/b5fec18c9bfec606e6c667e1c40cfe5f6abecb8e9fc968df8002321c02f4cbbf/diff",
  "WorkDir": "/var/lib/docker/overlay2/b5fec18c9bfec606e6c667e1c40cfe5f6abecb8e9fc968df8002321c02f4cbbf/work"
}
```

```
 ls /var/lib/docker/overlay2/b5fec18c9bfec606e6c667e1c40cfe5f6abecb8e9fc968df8002321c02f4cbbf/diff/root/.mc/
certs  config.json  session  share
```

**Reply by 160610101** [reply]
Thank you very much for your professional answer, but I still wonder why does this happen instead of directly existing in `~/.mc/config.json`?  Is it because I did something wrong? Or the README.md is out of date?
Thank u sincerely!

**Reply by harshavardhana** [reply]
> Thank you very much for your professional answer, but I still wonder why does this happen instead of directly existing in `~/.mc/config.json`? Is it because I did something wrong? Or the README.md is out of date?

docker container runs as root user, and the `.mc/config/json` exists at `${HOME}/.mc/`  of the user.
